### PR TITLE
Add otherProps util and DangerouslySetHTML component

### PR DESCRIPTION
**Pull request by bryceosterhaus**
From discussion on https://github.com/metal/metal.js/issues/239

Fixes https://github.com/metal/metal.js/issues/127
Fixes https://github.com/metal/metal.js/issues/205

### CODE REVIEW

**Comment by bryceosterhaus** [comment]
Not sure why this is failing on travis...

**Comment by jbalsas** [comment]
I've relaunched the tests, seemed like a network issue or something like that, all systems are green ☺️ 

I'll take a look at this on monday, but it looks good on first sight. In the meantime, maybe @mthadley  or @ethib137 who might be more familiar with the use case could chime in!

File: packages/metal-jsx/src/DangerouslySetHTML.js
Comment (mthadley) [inline packages/metal-jsx/src/DangerouslySetHTML.js]: Indentation looks a little funny here.

File: packages/metal-jsx/src/otherProps.js
Comment (mthadley) [inline packages/metal-jsx/src/otherProps.js]: I'm not sure if this really needs to change, but I can see most of the core library mostly prefers the C-style `for` loop over `forEach`. I'm guessing for performance? (The overhead of a function invocation for `n` items).

File: packages/metal-jsx/test/otherProps.js
Comment (mthadley) [inline packages/metal-jsx/test/otherProps.js]: I wonder if we can maybe start writing new tests completely in ES6? I mean, we are using `class` keywords already so...

**Review by mthadley — APPROVED** [review]
@jbalsas I added a few nitpicks, but all in all, this seems good. Thanks @bryceosterhaus!

**Comment by mthadley** [comment]
@bryceosterhaus Oh, I just realized something. Maybe this could be a class method on `JSXComponent`?

```js
class MyComponent extends JSXComponent {
  render() {
    return <div ...this.otherProps() class="content" />;
  }
}
```

**Comment by bryceosterhaus** [comment]
@mthadley, not a bad idea. Seems a little strange at first glance, but probably just because I am used to it being a separate function.

Also, for es6 in the unit tests, I decided to just follow the same patterns we had in our other tests

**Comment by mthadley** [comment]
@bryceosterhaus I figured you were, which is fine. But maybe going forward we can start pushing for more idiomatic ES6. I'll defer to you and @jbalsas on this.

**Comment by jbalsas** [comment]
Hey guys, thanks for working on this!

If `otherProps` is always meant to be used in the context of a `JSXComponent`, I also think it makes sense to make it part of the component directly, at least for now.

@bryceosterhaus, can you make that change and let me know? Regarding the tests, I'd love to start writing them in ES6 moving forward, but I wouldn't hold this PR for that, so no need to rewrite them right now.

Thanks!

**Comment by bryceosterhaus** [comment]
@jbalsas @mthadley, just updated pr so that otherProps is on the component itself

**Comment by jbalsas** [comment]
LGTM! @mthadley, one final check and I'll merge 😉

**Comment by mthadley** [comment]
🎉  @jbalsas LGTM  🎉

**Comment by eduardolundgren** [comment]
LGTM

**Comment by eduardolundgren** [comment]
Could you add a test case for using `DangerouslySetHTML` from inside another component?

**Comment by bryceosterhaus** [comment]
@eduardolundgren, added additional test case [here](https://github.com/metal/metal.js/pull/242/files#diff-d66c4d98a3d0bccc5855c5b99233541bR32)

**Comment by ethib137** [comment]
LGTM

**Pull request merged**
